internal/daemon/cluster/handlers/worker_service_test.go

```diff
@@ -1316,17 +1316,13 @@ func TestRoutingInfo(t *testing.T) {
 		require.NoError(t, err)
 
 		assert.Equal(t, w1.Address, "2.3.4.5:8080")
-		expTags := []*server.Tag{
-			{Key: "tag1", Value: "value1"},
-			{Key: "tag2", Value: "value2"},
+		expTags := server.Tags{
+			"tag1": []string{"value1"},
+			"tag2": []string{"value2"},
+		}
+		for k, v := range expTags {
+			assert.ElementsMatch(t, v, w1.ConfigTags[k])
```

Comment:
If we add an assertion that the length of `expTags` matches the length of `w1.ConfigTags`, then the tests would ensure there are not additional tag keys being returned.